/* ----------------------------------------------------------------------------
 * telemetry_kv -- Key-value GPB message definition
 *
 * November 2015, Jon Culver
 *
 * Copyright (c) 2015 by Cisco Systems, Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 * ----------------------------------------------------------------------------
 */

message Telemetry {
  optional uint64   collection_id = 1;
  optional string   base_path = 2;	
  optional string   subscription_identifier = 3;
  optional string   model_version = 4;
  optional uint64   collection_start_time = 5;
  optional uint64   msg_timestamp = 6;
  repeated TelemetryField fields = 14;
  optional uint64   collection_end_time = 15;
}

message TelemetryField {
  optional uint64         timestamp = 1;
  optional string         name = 2;
  optional bool           augment_data = 3;
  oneof value_by_type {
    bytes          bytes_value = 4;
    string         string_value = 5;
    bool           bool_value = 6;
    uint32         uint32_value = 7;
    uint64         uint64_value = 8;
    sint32         sint32_value = 9;
    sint64         sint64_value = 10;  
    double         double_value = 11;
    float          float_value = 12;
  }
  repeated TelemetryField fields = 15;
}

